api/proto/teleport/accesslist/v1/accesslist_service.proto

```diff
@@ -216,6 +216,15 @@ message UpsertAccessListWithMembersRequest {
   AccessList access_list = 1;
   // members is the list of access list members.
   repeated Member members = 2;
+  // preserve_expiry indicates that the upsert operation should preserve the
+  // existing membership expiry (if any) when overwriting an AccessList member
+  bool preserve_expiry = 3;
+  // preserve_reason indicates that the upsert operation should preserve the
+  // existing membership reason (if any) when overwriting an AccessList member
+  bool preserve_reason = 4;
+  // preserve_added_by indicates that the upsert opertation should preserve the
+  // existing "added_by" value
+  bool preserve_added_by = 5;
```

Comment:
Addig bool flag is a bit unclean solution

What is the use case/scenario when we don't want to preserve expiry, reason ,added by fields ? 


Right now we don't provide any mechanism (aport removing member and readding) that requires to modify this fields.  So why we just implement this as default mechanism.

Reply:
After doing a bit of reading, I don't think we have any ose cases where we edit an access list member at the moment - its delete/re-create if anything. Making it preserve the metadata by default seems like a sensible default.

(We just have to remember what it does if we want to allow live edits)